diff --git a/bad/libuser.py b/bad/libuser.py
--- a/bad/libuser.py
+++ b/bad/libuser.py
@@ -22,7 +22,7 @@
     conn = sqlite3.connect('db_users.sqlite')
     c = conn.cursor()
 
-    c.execute("INSERT INTO users (username, password, failures, mfa_enabled, mfa_secret) VALUES ('%s', '%s', '%d', '%d', '%s')" %(username, password, 0, 0, ''))
+    c.execute("INSERT INTO users (username, password, failures, mfa_enabled, mfa_secret) VALUES (?, ?, ?, ?, ?)", (username, password, 0, 0, ''))
 
     conn.commit()
     conn.close()
