diff --git a/bad/db.py b/bad/db.py
--- a/bad/db.py
+++ b/bad/db.py
@@ -16,7 +16,7 @@
     c.execute("CREATE TABLE users (user text, password text, failures int)")
 
     for u,p in users:
-        c.execute("INSERT INTO users (user, password, failures) VALUES ('%s', '%s', '%d')" %(u, p, 0))
+        c.execute("INSERT INTO users (user, password, failures) VALUES (?, ?, ?)", (u, p, 0))
 
     conn.commit()
     conn.close()
